Mathlib/Analysis/InnerProductSpace/Dual.lean

```diff
@@ -69,6 +70,61 @@ variable {E}
 theorem toDualMap_apply {x y : E} : toDualMap 𝕜 E x y = ⟪x, y⟫ :=
   rfl
 
+section NullSubmodule
+
+open SeparationQuotient LinearMap
+
+variable (E)
+
+/-- The null space with respect to the norm. -/
+def nullSubmodule : Submodule 𝕜 E where
+  __ := nullSubgroup
+  smul_mem' c x (hx : ‖x‖ = 0) := show ‖c • x‖ = 0 from
+    le_antisymm (norm_smul_le _ _ |>.trans <| by rw [hx, mul_zero]) (norm_nonneg _)
+
+@[simp]
+lemma mem_nullSubmodule_iff {x : E} : x ∈ nullSubmodule 𝕜 E ↔ ‖x‖ = 0 := Iff.rfl
+
+lemma inner_eq_zero_of_left (x y : E) (h : ‖x‖ = 0) :
+    ⟪x, y⟫_𝕜 = 0 := by
```

Comment:
I think this belongs closer to the Cauchy-Schwarz inequality, as it's not specific to the `nullSubmodule`. Same for the `of_right` version. And `x` should be implicit because it's capable of being inferred from `h`.
```suggestion
lemma inner_eq_zero_of_left {x : E} (y : E) (h : ‖x‖ = 0) :
    ⟪x, y⟫_𝕜 = 0 := by
```